packages/headless/src/controllers/product-listing/facet/headless-product-listing-facet.ts

```diff
@@ -167,6 +177,84 @@ export function buildFacet(
   };
 }
 
+export interface Facets extends Controller {
+  state: FacetsState;
+}
+
+export interface FacetsState {
+  facets: AutomaticFacet[];
+}
+
+// TODO(nico): We might need to extract a "core" automatic facets controller
```

Comment:
For everything related to automatic facets, I would not worry too much about it for the short term... 

I think it's needlessly complicating the first version of what we are trying to achieve here.

We should definitely support it eventually, but I would not put that in scope for the first few versions.

ie: Just having "normal" static facet working correctly from A to Z is a good enough achievement 😉

Reply:
With the PLP v2 endpoint, facets are effectively always automatic facets. A customer won't be able to specify which facets they want as the PLP v2 endpoint will return facets on the product listing fetch query. Once the facets are returned from the API, customers will be able to manipulate facets as usual.

This is why the facets are implemented as a variation of automatic facets, and not as regular facets.

Reply:
I see, I stand corrected then. Disregard the above !